src/librustc_lint/builtin.rs

```diff
@@ -57,24 +57,8 @@ use log::debug;
 // hardwired lints from librustc
 pub use lint::builtin::*;
 
-declare_lint! {
-    WHILE_TRUE,
-    Warn,
-    "suggest using `loop { }` instead of `while true { }`"
-}
-
-#[derive(Copy, Clone)]
-pub struct WhileTrue;
-
-impl LintPass for WhileTrue {
-    fn name(&self) -> &'static str {
-        "WhileTrue"
-    }
-
-    fn get_lints(&self) -> LintArray {
-        lint_array!(WHILE_TRUE)
-    }
-}
+declare_lint!(WHILE_TRUE, Warn, "suggest using `loop { }` instead of `while true { }`");
```

Comment:
please leave the `declare_lint` invocations as they were. It's become the de-facto style and often it can't be changed anyway due to the length of the message.

Reply:
I changed it back, but note that you can see in this PR that there are places I didn't changed that use this style. I also think having `declare_lint!(` and `declare_lint_pass!` neatly together on two lines like this is quite readable.

Reply:
hmm. maybe I'm just so conditioned to clippy's use of `declare_lint`. Having the three components be in three different lines allows me to immediately see the entire lint instead of having to parse a line.